multiqc/multiqc.py

```diff
@@ -362,33 +395,195 @@ def run(
+        plots_flat=plots_flat,
+        plots_interactive=plots_interactive,
+        strict=strict,
+        lint=lint,
+        development=development,
+        make_pdf=make_pdf,
+        no_megaqc_upload=no_megaqc_upload,
+        config_file=config_file,
+        cl_config=cl_config,
+        profile_runtime=profile_runtime,
+        no_ansi=no_ansi,
+        custom_css_files=custom_css_files,
+        **kwargs,
+    )
+
+    try:
+        run_modules, run_module_names = _file_search(
+            file_list=file_list,
+            ignore=ignore,
+            ignore_samples=ignore_samples,
+        )
+
+        # Create the temporary working directories
+        tmp_dir = tempfile.mkdtemp()
+        logger.debug(f"Using temporary directory for creating report: {tmp_dir}")
+
+        # Load the template
+        template_mod = config.avail_templates[config.template].load()
+
+        _run_modules(
+            tmp_dir=tmp_dir,
+            template_mod=template_mod,
+            filename=filename,
+            run_modules=run_modules,
+            run_module_names=run_module_names,
+        )
+
+        _general_stats_table()
+
+        _export_sources()
+
+        _write_json_dump()
+
+        _write_html_and_data(
+            tmp_dir=tmp_dir,
+            template_mod=template_mod,
+            filename=filename,
+        )
+
+    except _RunError as e:
+        if e.message:
+            logger.critical(e.message)
+        return RunResult(message=e.message, sys_exit_code=e.sys_exit_code)
+
+    plugin_hooks.mqc_trigger("execution_finish")
+
+    report.runtimes["total"] = time.time() - start_execution_time
+    if config.profile_runtime:
+        logger.warning(f"Run took {report.runtimes['total']:.2f} seconds")
+        logger.warning(f" - {report.runtimes['total_sp']:.2f}s: Searching files")
+        logger.warning(f" - {report.runtimes['total_mods']:.2f}s: Running modules")
+        if config.make_report:
+            logger.warning(f" - {report.runtimes['total_compression']:.2f}s: Compressing report data")
+            logger.info(f"For more information, see the 'Run Time' section in {os.path.relpath(config.output_fn)}")
+
+    if report.num_mpl_plots > 0 and not config.plots_force_flat:
+        if not config.plots_force_interactive:
+            console.print(
+                "[blue]|           multiqc[/] | "
+                "Flat-image plots used. Disable with '--interactive'. "
+                "See [link=https://multiqc.info/docs/#flat--interactive-plots]docs[/link]."
+            )
+
+    sys_exit_code = 0
+    if config.strict and len(report.lint_errors) > 0:
+        logger.error(f"Found {len(report.lint_errors)} linting errors!\n" + "\n".join(report.lint_errors))
+        sys_exit_code = 1
+
+    logger.info("MultiQC complete")
+
+    # Move the log file into the data directory
+    log.move_tmp_log(logger)
+
+    return RunResult(sys_exit_code=sys_exit_code)
+
+
+def _set_up_logging(verbose, quiet, no_ansi):
```

Comment:
Might make sense to move this to [`multiqc/utils/log.py`](https://github.com/MultiQC/MultiQC/blob/main/multiqc/utils/log.py) - there's already the `init_log` function there.

Reply:
Yep, I've actually done that later in https://github.com/MultiQC/MultiQC/pull/2442

Reply:
Ok cool - good to merge then 👍🏻